app/acme/dnsprovider/cloudns.go

```diff
@@ -0,0 +1,289 @@
+package dnsprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	envPrefix = "CLOUDNS_"
```

Comment:
I think you restrict the setup of providers with env only? If this is the case we probably better use some simple config lib, allowing to set those in a config file (probably yml) and supporting env as well. I think I have seen such a thing

Reply:
yes, I only implemented setup thorough the environment variables. But  I fully agree that to have a config possibility would be very handy. What library do you mean?

Reply:
I didn't have any particular lib in mind, maybe this one? https://github.com/ilyakaznacheev/cleanenv 

Seems to support both yaml and toml as well as env